walk-in-interview/src/main/webapp/homepage/script.js

```diff
@@ -0,0 +1,233 @@
+function addJobOrderOptions() {
+  const jobOrderSelect = document.getElementById('homepage-sort-by-order');
+
+  addSelectOptions(jobOrderSelect, STRINGS['homepage-sort-by-order']);
+}
+
+/**
+ * Add the keys and values from the options map to the select element.
+ * @param {Element} select The select element.
+ * @param {Map} options The map of options to be added.
+ */
+function addSelectOptions(select, options) {
+  select.options.length = 0;
+
+  for (const key in options) {
+    if (options.hasOwnProperty(key)) {
+      select.options[select.options.length] =
+        new Option(options[key], key);
+    }
+  }
+}
+
+/**
+ * Add the attributes and on click function to the sorting
+ * submit button.
+ */
+function addJobSortSubmit() {
+  const sortBySubmit = document.getElementById('homepage-sort-by-submit');
+  sortBySubmit.setAttribute('type', 'submit');
+  sortBySubmit.setAttribute('value', STRINGS['homepage-sort-by-submit']);
+
+  sortBySubmit.addEventListener('click', (_) => {
+    if (!validSortByInput()) {
+      return;
+    }
+
+    const sortByParam = document.getElementById('homepage-sort-by').value;
+    const sortOrderParam = document.getElementById('homepage-sort-by-order');
+    addJobListings(sortByParam, sortOrderParam, DEFAULT_PAGE_SIZE,
+        DEFAULT_PAGE_INDEX);
+  });
+}
+
+/**
+ * Add the attributes and on click function to the filters
+ * submit button.
+ */
+function addJobFilterSubmit() {
+  const filterBySubmit = document.getElementById('homepage-filter-by-submit');
+  filterBySubmit.setAttribute('type', 'submit');
+  filterBySubmit.setAttribute('value', STRINGS['homepage-filter-by-submit']);
+
+  /**
+   * TODO(issue/35): add filtering event listener
+   * + validation + GET request functions
+   */
+}
+
+/**
+ * Add the list of jobs that are stored in the database.
+ * @param {String} sortBy How the jobs should be sorted.
+ * @param {String} order The order of the sorting.
+ * @param {int} pageSize The number of jobs for the page.
+ * @param {int} pageIndex The page index (starting from 0).
+ */
+function addJobListings(sortBy, order, pageSize, pageIndex) {
+  const jobListings = getJobListings(sortBy, order, pageSize, pageIndex);
+  console.log('jobListings', jobListings);
+  const jobListingsElement = document.getElementById('job-listings');
+
+  jobListingsElement.innerHTML = '';
+  const jobListingTemplate = document.getElementById('job-listing-template');
+
+  jobListings.forEach((job) => {
+    const jobListing =
+      jobListingTemplate.cloneNode( /** and child elements */ true);
+    jobListing.setAttribute('id', job['jobId']);
+
+    const jobTitle = jobListing.children[0];
+    jobTitle.innerText = job['jobTitle'];
+
+    const jobAddress = jobListing.children[1];
+
+    const location = job['jobLocation'];
+    const address = location['address'];
+    const postalCode = location['postalCode'];
+    jobAddress.innerText = address + ', ' + postalCode;
```

Comment:
suppose javascript has something like `format` in java to format the string and increase readability

Reply:
yep, will use ES6 as follow: `${address}, ${postalCode}`